buildSrc/src/main/java/org/opensearch/gradle/util/GradleUtils.java

```diff
@@ -254,21 +249,12 @@ public static String getProjectPathFromTask(String taskPath) {
         return lastDelimiterIndex == 0 ? ":" : taskPath.substring(0, lastDelimiterIndex);
     }
 
-    public static FileCollection getFiles(Project project, Configuration cfg, Spec<Dependency> spec) {
-        final LenientConfiguration configuration = cfg.getResolvedConfiguration().getLenientConfiguration();
-        try {
-            // Using reflection here to cover the pre 8.7 releases (since those have no such APIs), the
-            // ResolverResults.LegacyResolverResults.LegacyVisitedArtifactSet::select(...) is not available
-            // on older versions.
-            final MethodHandle mh = MethodHandles.lookup()
-                .findVirtual(configuration.getClass(), "select", MethodType.methodType(SelectedArtifactSet.class, Spec.class))
-                .bindTo(configuration);
+    public static FileCollection getFiles(Project project, Configuration cfg, Spec<? super ComponentIdentifier> spec) {
```

Comment:
@reta if at any point you have a few minutes I would appreciate some help here. I'm able to get `./gradlew precommit -x dependencyLicenses -x thirdPartyAudit` to succeed, but dependencyLicenses and thirdPartyAudit are the two tasks that use this method and they fail. I've tried to switch over to use [artifact views](https://docs.gradle.org/current/userguide/artifact_views.html) here but obviously am doing something wrong.